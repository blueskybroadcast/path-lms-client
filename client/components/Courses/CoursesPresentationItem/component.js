import React from 'react';
import PropTypes from 'prop-types';

const CoursesPresentationItem = ({
  coverDescription, coverPhotoUrl, id, name
}) => (
  <article className="course">
    <a title="Online Course" href={`/showroom/courses/${id}`}>
      <img
        className="course-cover-photo"
        alt=""
        src={coverPhotoUrl}
      />
      <h3 data-role="course-info" title="Online Course">
        {name}
        <span
          className="status icon icon-purchasable-usd unpaid"
          aria-label="Unpaid"
        />
        <span
          className="status icon hidden"
          aria-label=""
        />
      </h3>
      <div className="info">
        <p>{coverDescription}</p>
      </div>
      <footer className="meta-info">
        <ul>
          <li>
            <i className="icon icon-section" />
            <span>3</span>
          </li>
          <li>
            <i className="icon icon-presentation" />
            <span>2</span>
          </li>
          <li>
            <i className="icon icon-link" />
            <span>3</span>
          </li>
          <li>
            <i className="icon icon-assessment" />
            <span>2</span>
          </li>
          <li>
            <i className="icon icon-document" />
            <span>1</span>
          </li>
          <li>
            <i className="icon icon-certificate" />
            <span>1</span>
          </li>
          <li>
            <i className="icon icon-survey" />
            <span>3</span>
          </li>
          <li>
            <i className="icon icon-assignment" />
            <span>1</span>
          </li>
          <li>
            <i className="icon icon-in-person-event" />
            <span>1</span>
          </li>
        </ul>
      </footer>
    </a>
  </article>
);

CoursesPresentationItem.propTypes = {
  coverDescription: PropTypes.string,
  coverPhotoUrl: PropTypes.string,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

CoursesPresentationItem.defaultProps = {
  coverDescription: null,
  coverPhotoUrl: null
};

export default CoursesPresentationItem;
