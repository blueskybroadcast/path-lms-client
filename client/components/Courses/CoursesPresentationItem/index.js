import CoursesPresentationItem from './component';

export default CoursesPresentationItem;
