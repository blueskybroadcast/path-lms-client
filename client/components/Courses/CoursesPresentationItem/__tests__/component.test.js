import React from 'react';
import { shallow } from 'enzyme';
import CoursesPresentationItem from '../component';

describe('Courses Presentation Item', () => {
  const props = {
    id: '56',
    name: 'Presentation Item name',
    slug: 'testing',
    currency: 'usd'
  };
  test('should render CoursesPresentationItem with minimum data correctly', () => {
    const wrapper = shallow(
      <CoursesPresentationItem {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render CoursesPresentationItem with all data correctly', () => {
    const wrapper = shallow(
      <CoursesPresentationItem
        {...props}
        coverDescription="Description"
        resizedCoverPhotoUrl="coverPhotoUrl"
        permittedContent={{
          assessment: 2,
          assignment: 1,
          certificate: 3,
          document: 1,
          link: 5,
          presentation: 9,
          section: 6,
          survey: 2
        }}
        statuses={{ purchase: 'unpaid' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
