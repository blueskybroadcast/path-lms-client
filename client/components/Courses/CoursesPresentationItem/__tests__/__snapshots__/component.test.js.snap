// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Courses Presentation Item should render CoursesPresentationItem with all data correctly 1`] = `
<article
  className="course"
>
  <a
    href="/testing/courses/56"
    title="Online Course"
  >
    <img
      alt=""
      className="course-cover-photo"
      src="coverPhotoUrl"
    />
    <h3
      data-role="course-info"
      title="Online Course"
    >
      Presentation Item name
      <span
        className="status icon icon-purchasable-usd unpaid"
      />
    </h3>
    <div
      className="info"
    >
      <p>
        Description
      </p>
    </div>
    <footer
      className="meta-info"
    >
      <ul>
        <MetaItem
          count={6}
          icon="section"
        />
        <MetaItem
          count={9}
          icon="presentation"
        />
        <MetaItem
          count={5}
          icon="link"
        />
        <MetaItem
          count={1}
          icon="document"
        />
        <MetaItem
          count={3}
          icon="certificate"
        />
        <MetaItem
          count={2}
          icon="survey"
        />
        <MetaItem
          count={1}
          icon="assignment"
        />
      </ul>
    </footer>
  </a>
</article>
`;

exports[`Courses Presentation Item should render CoursesPresentationItem with minimum data correctly 1`] = `
<article
  className="course"
>
  <a
    href="/testing/courses/56"
    title="Online Course"
  >
    <img
      alt=""
      className="course-cover-photo"
      src={null}
    />
    <h3
      data-role="course-info"
      title="Online Course"
    >
      Presentation Item name
      <span
        className="status icon icon-purchasable-usd"
      />
    </h3>
    <div
      className="info"
    >
      <p />
    </div>
    <footer
      className="meta-info"
    >
      <ul />
    </footer>
  </a>
</article>
`;
